### ISSUE TITLE
Calling client.get_torrent_list results in a panick

### ISSUE THREAD

**Issue by DestinyofYeet** [George-Miao/qbit#7]
Calling
```rust
client.get_torrent_list(GetTorrentListArg::default()).await.unwrap();
```
Results in a panick

[log.txt](https://github.com/George-Miao/qbit/files/13467798/log.txt)

**Comment by George-Miao** [comment]
Try request the API manually and check if the result is what you would expect (list of torrents in json).

**Comment by DestinyofYeet** [comment]
It does work with other libraries, so I don't think the issue is on my end. Tested with https://github.com/Koro33/qbit-api-rs

**Comment by George-Miao** [comment]
Please provide more info, include qbittorrent version, qbit-rs version, and the test code you used.

**Comment by DestinyofYeet** [comment]
Sorry, took a while.

`qbit-rs = "0.3.8"`
qBittorrent v4.2.5

test code:
```rust
use qbit_rs::model::{Credential, GetTorrentListArg};
use qbit_rs::Qbit;

#[tokio::main]
pub async fn main() {
    let client = Qbit::new("http://192.168.0.252:8080", Credential::new("xxx", "xxx"));

    client.login(true).await.unwrap();

    println!("Logged in!");

    let torrent_list = client.get_torrent_list(GetTorrentListArg::default()).await.unwrap();
}
```

**Comment by George-Miao** [comment]
Ok I know what's going on. In old version, `torrents/info` (maybe other "read" endpoints) only accepts GET with url param, while in new versions we can do a post request with JSON body. I'll fix this soon.

**Comment by George-Miao** [comment]
I have fixed the bug and tested with qbittorrent v4.2.5 without problem anymore. Please upgrade qbit-rs version to 0.4 and try again.

**Comment by DestinyofYeet** [comment]
Unfortunately, I'm now getting a 

```
thread 'main' panicked at src/main.rs:8:30:
called `Result::unwrap()` on an `Err` value: BadResponse { explain: "Failed to extract cookie from response" }
```

with the same example

`qbit-rs = "0.4.1"`

[rust_error.txt](https://github.com/George-Miao/qbit/files/13504978/rust_error.txt)


_Edit:_

I'mma try to reproduce it at home, since I was using a vpn, but I don't think it should make a difference.

Another edit:

Nevermind, seems like the vpn was the issue
